Support fully qualified bucket paths

arempter opened the pull request:
- allows S3 plugin to search for all objects in S3 buckets (also pseudodirectories)
- allows S3 plugin to connect via Airlock with NPA user

arempter committed:
s3 plugin refactor to use aws sdk only

arempter committed:
fix test

arempter committed:
lower signature version type to pass airlock proxy

arempter committed:
Readme update

arempter committed:
comments cleanup

arempter committed:
allow to configure aws region via parameters

arempter committed:
fix test dependencies

nielsdenissen commented on pom.xml:
```diff
@@ -31,11 +31,17 @@
         <ranger.version>1.1.0</ranger.version>
-        <aws.version>1.11.285</aws.version>
+        <aws.version>1.11.461</aws.version>
         <radosgw.version>1.0.2</radosgw.version>
```
We wouldn't need radosgw anymore right?

nielsdenissen commented on README.md:
```diff
@@ -36,3 +36,35 @@ http://{RANGER_HOST}:{RANGER_PORT}/service/public/v2/api/servicedef
+If you run plugin via Airlock, make sure that:
+
+- user used in S3 service setup is NPA user in [Airlock STS](https://github.com/ing-bank/airlock-sts)
+- user is added to all - path Ranger policy 
```
What do you mean here?

nielsdenissen reviewed (changes requested):
Currently, only buckets are autocompleted by Ranger (as we only implement a `getBucket` function in the `lookupResource` function of `RangerServiceS3.java`). This feature ensures fewer typos happen.

For this I'd propose a solution where while the user is typing, the next level will be queried and shown to him. So:
- when typing `demob`: suggestions could be `demobucket` and `demobucket2`
- when typing `demobucket/`: suggestions could be `demobucket/subdir1` and `demobucket/subdir2`
- when typing `demobucket/subdir1/`: suggestions could be `demobucket/subdir1/whatever`
- etc...

Reasons for this being it will remove error-prone manual typing and confirm to the user this folder exists. Showing the users all nested options could become far too large though, this would help the navigation

arempter committed:
disable virtual hosts resolution (auto detect) in aws sdk

arempter committed:
changed behavior of subdir search
- no files are returned
- only 50 first matches
- subdir is only searched after putting / in bucket name

arempter committed:
fix tests after search change

arempter committed:
subdirectory search update

arempter committed:
subdirectory search update

arempter committed:
update of subfolder parsing

arempter committed:
final udpates on readme, typos, formatting
